src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,157 @@
+package duke.storage;
+
+import duke.exception.ChiException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.tasklist.TaskList;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Loads and stores tasks from given data file to and from application.
+ */
+public class Storage {
+
+    /** The relative file path of the data file */
+    private String dataFilePath;
+
+    /** The relative path of the directory containing the data file */
+    private String dataFolderPath;
+
+    /** The File object representing the data file */
+    private File dataFile;
+
+    /**
+     * Constructor of the class.
+     *
+     * @param path The relative path of the data file to use for storage of tasks.
+     */
+    public Storage(String path) {
+        this.dataFilePath = path;
+        String[] rPath = path.split("/");
+        StringBuilder s = new StringBuilder("./");
+        for (int i = 0; i < rPath.length - 1; i++) {
+            s.append(rPath[i]);
+        }
+        this.dataFolderPath = s.toString();
+    }
+
+    /**
+     * Deciphers a task in the data file.
+     *
+     * @param task A single task stored in the data file.
+     * @return A Task instance of the task read from the data file.
+     */
+    private Task convertFileDataToTask(String task) {
+        // We assume the task stored inside is correct
+        String[] splitTask = task.split("\\|");
+        for (int i = 0; i < splitTask.length; i++) {
+            splitTask[i] = splitTask[i].trim();
+        }
+        if (splitTask[0].equals("T")) {
+            if (splitTask[1].equals("1")) {
+                return new Todo(splitTask[2], true);
+            }
+            return new Todo(splitTask[2], false);
+        } else if (splitTask[0].equals("D")) {
+            if (splitTask.length == 5) {
+                return new Deadline(splitTask[2], LocalDate.parse(splitTask[3],
+                        DateTimeFormatter.ofPattern("yyyy-MM-dd")),
+                        LocalTime.parse(splitTask[4], DateTimeFormatter.ofPattern("HH:mm")),
+                        splitTask[1].equals("1"));
+            }
+            /*else {
```

Comment:
Maybe it might be better to remove the commented out else statement if it's not in use anymore?